src/App.tsx

```diff
@@ -17,38 +47,39 @@ export const App: React.FC = () => {
             type="text"
             placeholder="Enter a part of the name"
             className="input"
+            value={isInput}
+            onChange={handleInputChange}
+            onFocus={() => setInputFocused(true)}
+            onBlur={() => setInputFocused(false)}
           />
         </div>
 
         <div className="dropdown-menu" role="menu">
           <div className="dropdown-content">
-            <div className="dropdown-item">
-              <p className="has-text-link">Pieter Haverbeke</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-link">Pieter Bernard Haverbeke</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-link">Pieter Antone Haverbeke</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-danger">Elisabeth Haverbeke</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-link">Pieter de Decker</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-danger">Petronella de Decker</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-danger">Elisabeth Hercke</p>
-            </div>
+            {isInputFocused && filteredPeople.length > 0 ? (
+              filteredPeople.map(person => (
+                <div
+                  className="dropdown-item"
+                  key={person.name}
+                  onMouseDown={() => handleSelected(person)}
+                  onKeyDown={(e) => {
+                    if (e.key === 'Enter') {
+                      handleSelected(person);
+                    }
+                  }}
```

Comment:
Move to separate function  (transfer the function to a variable, and here only write the name of the function)